File: vosk-prototype/Tool.py
class Tool:
    def __init__(self, tool_names, tool_handle):
        self.tool_names = tool_names
        self.tool_handle = tool_handle

    def search_property(self, speech_transcription, tool_property_search):
        self.searchname(speech_transcription, tool_property_search)

    def searchname(self, speechTranscription, tool_property_search):
        for name in self.tool_names:
            if speechTranscription.find(name) != -1:
                tool_property_search.tool_handle_ = self.tool_handle
                return True
        return False

    def searcholdname(self, tool_property_search):
        if tool_property_search.toolneedsize_:
            return self.searchname(tool_property_search.toolneedsize_, tool_property_search)
        if tool_property_search.screw_spine_property_search_.screw_need_diameter_:
            return self.searchname(tool_property_search.screw_spine_property_search_.screw_need_diameter_,
                                   tool_property_search)
        if tool_property_search.screw_spine_property_search_.screw_need_length_:
            return self.searchname(tool_property_search.screw_spine_property_search_.screw_need_length_,
                                   tool_property_search)
        if tool_property_search.screw_wrist_property_search_.screw_need_length_:
            return self.searchname(tool_property_search.screw_wrist_property_search_.screw_need_length_,
                                   tool_property_search)
        if tool_property_search.screw_wrist_property_search_.screw_need_headType_:
            return self.searchname(tool_property_search.screw_wrist_property_search_.screw_need_headType_,
                                   tool_property_search)
        if tool_property_search.screw_wrist_property_search_.screw_need_thread_fullness_:
            return self.searchname(tool_property_search.screw_wrist_property_search_.screw_need_thread_fullness_,
                                   tool_property_search)
        if tool_property_search.screw_ankle_property_search_.screw_need_length_:
            return self.searchname(tool_property_search.screw_ankle_property_search_.screw_need_length_,
                                   tool_property_search)
        if tool_property_search.screw_ankle_property_search_.screw_need_thread_type_:
            return self.searchname(tool_property_search.screw_ankle_property_search_.screw_need_thread_type_,
                                   tool_property_search)
        return False

File: Microsoft-prototype/Tool.cpp

#include "Tool.h" 
#include <iostream>
using namespace std;

Tool::Tool(const std::vector<string>& toolNames, const ToolHandleType& toolHandle) : toolNames_(toolNames), toolHandle_(toolHandle) {
}

void Tool::searchProperty(const string& speechTranscription, ToolPropertySearch& toolPropertySearch)const {
	this->searchName(speechTranscription, toolPropertySearch);
}

bool Tool::searchName(const string& speechTranscription, ToolPropertySearch& toolPropertySearch)const {
	for (const auto& name : toolNames_) {
		if (speechTranscription.find(name) != string::npos) {
			toolPropertySearch.toolHandle = toolHandle_;
			return true;
		}
	}
	return false;

}


bool Tool::searchOldName(ToolPropertySearch& toolPropertySearch) const {
	if (!toolPropertySearch.toolNeedSize.empty()) {
		return searchName(toolPropertySearch.toolNeedSize, toolPropertySearch);
	}
	if (!toolPropertySearch.screwSpinePropertySearch.screwNeedDiameter.empty()) {
		return searchName(toolPropertySearch.screwSpinePropertySearch.screwNeedDiameter, toolPropertySearch);
	}
	if (!toolPropertySearch.screwSpinePropertySearch.screwNeedLength.empty()) {
		return searchName(toolPropertySearch.screwSpinePropertySearch.screwNeedLength, toolPropertySearch);
	}
	if (!toolPropertySearch.screwWristPropertySearch.screwNeedHeadType.empty()) {
		return searchName(toolPropertySearch.screwWristPropertySearch.screwNeedHeadType, toolPropertySearch);
	}
	if (!toolPropertySearch.screwWristPropertySearch.screwNeedThreadFullness.empty()) {
		return searchName(toolPropertySearch.screwWristPropertySearch.screwNeedThreadFullness, toolPropertySearch);
	}
	if (!toolPropertySearch.screwWristPropertySearch.screwNeedLength.empty()) {
		return searchName(toolPropertySearch.screwWristPropertySearch.screwNeedLength, toolPropertySearch);
	}

	if (!toolPropertySearch.screwAnklePropertySearch.screwNeedLength.empty()) {
		return searchName(toolPropertySearch.screwAnklePropertySearch.screwNeedLength, toolPropertySearch);
	}
	if (!toolPropertySearch.screwAnklePropertySearch.screwNeedThreadType.empty()) {
		return searchName(toolPropertySearch.screwAnklePropertySearch.screwNeedThreadType, toolPropertySearch);
	}
	return false;
}







